src/reference-manual/user-functions.Rmd

```diff
@@ -36,10 +36,42 @@ real foo(real mu, real sigma);
 
 declares a function named `foo` with two argument variables of
 types `real` and `real`.  The arguments are named `mu`
-and `sigma`, but that is not part of the declaration.  Two
-user-defined functions may *not* have the same name even if they have
-different sequences of argument types.
+and `sigma`, but that is not part of the declaration.
 
+### Function overloading
+Two user-defined functions may have the same name if they have
+different sequences of [argument types](#argument-types-and-qualifiers).
+
+This is known as function overloading.
+
+For example, the following two functions are both defined with the name `add_up`
+
+```stan
+real add_up(real a, real b){
+  return a + b;
+}
+
+real add_up(real a, real b, real c){
+  return a + b + c;
+}
+```
+
+The return types of overloaded functions do not need to be the same as a matter
+of course, but it may be confusing if they differ.
```

Comment:
We don't have dependent types, so we'll run into cases most naturally coded with differing returns, as in 

```
int add_up(int a, int b) { return a + b; }
```